CTCP-960 : Transit navigation logic

jackheslop96 opened the pull request:
- Navigators fetch the reference data lists as these are required for the user answers reader
- Hide remove link on first office of transit

ajchilts reviewed (dismissed):
Massive effort here! 👍 

On the face of it the solution looks good. Maybe some small refactoring we could reduce complexity in some of the larger functions but on the whole it looks good. 

One suggestion is to leave comments or TODOs where we are referencing endpoints or ref data that may change moving forward? 


Great work @jackheslop96

jackheslop96 commented:
> Massive effort here! 👍
> 
> On the face of it the solution looks good. Maybe some small refactoring we could reduce complexity in some of the larger functions but on the whole it looks good.
> 
> One suggestion is to leave comments or TODOs where we are referencing endpoints or ref data that may change moving forward?
> 
> Great work @jackheslop96

Thanks @ajchilts. Turns out I was using the wrong data list (CL009 instead of CL112). I have added in a new endpoint and updated this branch to point to it.
